secure-api-gateway-ob-uk-rs-resource-store/secure-api-gateway-ob-uk-rs-resource-store-api/src/main/java/com/forgerock/sapi/gateway/rs/resource/store/api/admin/events/DataEventsApi.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.rs.resource.store.api.admin.events;
+
+
+import java.util.Collection;
+
+import javax.validation.Valid;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import com.forgerock.sapi.gateway.ob.uk.common.error.OBErrorResponseException;
+import com.forgerock.sapi.gateway.rs.resource.store.datamodel.events.FREventMessages;
+
+import io.swagger.annotations.ApiParam;
+
+
+@RequestMapping("/admin/data")
+public interface DataEventsApi {
+
+    @RequestMapping(value = "/events",
+            produces = {"application/json; charset=utf-8"},
+            consumes = {"application/json; charset=utf-8"},
+            method = RequestMethod.POST)
+    ResponseEntity<FREventMessages> importEvents(
+            @ApiParam(value = "Default", required = true)
+            @Valid
+            @RequestBody FREventMessages frEventMessages
+    ) throws OBErrorResponseException;
+
+    @RequestMapping(value = "/events",
+            produces = {"application/json; charset=utf-8"},
+            consumes = {"application/json; charset=utf-8"},
+            method = RequestMethod.PUT)
+    ResponseEntity<FREventMessages> updateEvents(
+            @ApiParam(value = "Default", required = true)
+            @Valid
+            @RequestBody FREventMessages frEventMessages
+    ) throws OBErrorResponseException;
+
+    @RequestMapping(value = "/events/all",
+            produces = {"application/json; charset=utf-8"},
+            method = RequestMethod.GET
+    )
+    ResponseEntity<Collection<FREventMessages>> exportEvents();
+
+    @RequestMapping(value = "/events",
+            produces = {"application/json; charset=utf-8"},
+            method = RequestMethod.GET
+    )
+    ResponseEntity<FREventMessages> exportEventsByTppId(
+            @ApiParam(value = "Default", required = true)
+            @Valid
+            @RequestParam String tppId
+    );
```

Comment:
Results may need to be paged. 

Alternatively, as this is a sandbox then I think it would be good to enforce a configurable max number of events per ApiClient, then the need for paging would be avoided. It would also allow us to limit the max amount of database space required, and prevent a rogue ApiClient from creating an unreasonable amount of events.

Reply:
I agree with that solution to enforce a configurable max number of events per apiClient, at the end the events pushed by the api client will be only for test purposes, so that to apply a limit make sense to me.

Reply:
And avoid to implement a paged solution.

Reply:
Added import event limit, now the implementation check:
- The number of the events provided in the payload against the limit configured
- The number of the events stored, if the stored events reach the limit import new events are not allowed
- The total number of the events that will be stored, the current stored events plus the events provided in the request